src/core_modules/capture-core/metaDataMemoryStoreBuilders/programs/factory/enrollment/EnrollmentFactory.js

```diff
@@ -91,63 +85,138 @@ export class EnrollmentFactory {
         return DataElementFactory.buildtetFeatureType(featureType);
     }
 
-    async _buildSection(
-        cachedProgramTrackedEntityAttributes: ?Array<CachedProgramTrackedEntityAttribute>,
-        cachedProgramTrackedEntityTypeId: ?string,
+    async _buildTetFeatureTypeSection(
+        cachedProgramTrackedEntityTypeId: string,
     ) {
         const featureTypeField = this._buildTetFeatureTypeField(cachedProgramTrackedEntityTypeId);
-        if ((!cachedProgramTrackedEntityAttributes ||
-            cachedProgramTrackedEntityAttributes.length <= 0) &&
-            !featureTypeField) {
+        const trackedEntityTypeName = this.cachedTrackedEntityTypes.get(cachedProgramTrackedEntityTypeId);
+
+        if (!trackedEntityTypeName?.displayName || !featureTypeField) {
             return null;
         }
 
+        const section = new Section((o) => {
+            o.id = cachedProgramTrackedEntityTypeId;
+            o.name = trackedEntityTypeName.displayName;
+        });
```

Comment:
I'm going to nitpick a bit here 😇 
- `trackedEntityTypeName` is really `trackedEntityType`
- If for some reason the `trackedEntityType` is not found or doesn't have a `displayName`, I think we should still  show the section (without a name).